reckless_github/src/repository.rs

```diff
@@ -35,6 +37,13 @@ impl Repository for Github {
     /// Where to store the index is an implementation
     /// details.
     async fn init(&self) -> Result<(), RecklessError> {
+        let repo =
+            match git2::Repository::clone_recurse(&self.url, get_dir_path_from_url(&self.url)) {
+                Ok(repo) => repo,
+                Err(e) => panic!("failed to open: {}", e),
+            };
```

Comment:
I think this is very bugged and I will contact the team to understand why, for now this is a workaround

```diff
diff --git a/reckless_github/src/repository.rs b/reckless_github/src/repository.rs
index c95e403..0f219ef 100644
--- a/reckless_github/src/repository.rs
+++ b/reckless_github/src/repository.rs
@@ -37,14 +37,27 @@ impl Repository for Github {
     /// Where to store the index is an implementation
     /// details.
     async fn init(&self) -> Result<(), RecklessError> {
-        let repo =
-            match git2::Repository::clone_recurse(&self.url, get_dir_path_from_url(&self.url)) {
-                Ok(repo) => repo,
-                Err(e) => panic!("failed to open: {}", e),
-            };
+        let res = git2::Repository::clone(&self.url, get_dir_path_from_url(&self.url));
 
-        println!("{} was initialized", self.name);
-        Ok(())
+        match res {
+            Ok(repo) => {
+                // FIXME: put this inside a utils methods and call the method `clone_recursive_fix`.
+                for sub in repo.submodules().unwrap_or_default() {
+                    println!("URL {}", sub.url().unwrap());
+                    let path = format!(
+                        "{}/{}",
+                        get_dir_path_from_url(&self.url),
+                        sub.path().to_str().unwrap()
+                    );
+                    match git2::Repository::clone(sub.url().unwrap(), path) {
+                        Ok(_) => println!("Ok {:?}", sub.path()),
+                        Err(err) => println!("KO {:?} {:?}", sub.path(), err),
+                    }
+                }
+                Ok(())
+            }
+            Err(err) => Err(RecklessError::new(1, err.message())),
+        }
     }

```

Also with some rust improvement 😄